apheleia-formatters.el

```diff
@@ -239,11 +240,13 @@ rather than using this system."
     (cmake-mode . cmake-format)
     (cmake-ts-mode . cmake-format)
     (common-lisp-mode . lisp-indent)
+    (conf-toml-mode . dprint)
```

Comment:
dprint supports lots of things. I opted to add here only some of the most common associations that did not have any yet: toml, Dockerfile, markdown.

dprint also requires one to have a plugin installed for each type, so there's no guarantee that even these would work -- there might not be a plugin installed for these or a config in effect might not reference one, etc. Perhaps the safer bet would be to not add any here :thinking:

Reply:
I think what we want is probably some way to communicate to Apheleia whether or not the plugin is installed. Can that be checked quickly? Currently, Apheleia checks whether the binary to be executed for a formatter is on `$PATH`, and if it is not, then it silently will skip formatting, because of course trying to invoke it would fail. We could add a syntax that could be used to specify a custom check for whether the formatter exists.

Or, alternatively, we could use the existing facility of adding wrapper scripts like `apheleia-phpcs` and `apheleia-npx`, and pass the plugin name as an argument. The wrapper script would check its existence, silently return if the plugin is not available, and delegate to it otherwise.

Reply:
I'm not sure if that plugin existence can be checked that quickly/cleanly. Would need to traverse dirs upwards, locating one of the 4 different allowed filenames (following dprint's order of precedence), and then grabbing something out of the found file which can be json or jsonc, https://dprint.dev/setup/

But I found that dprint falls back to just passthrough without any formatting (like `cat`) if it gets passed a filename that is not handled by the installed plugins. I think with this behavior we could just add whatever associations we see appropriate and know are working with _some_ dprint+plugin config to the default `apheleia-mode-alist` -- without an applicable plugin, dprint wouldn't just change anything. I'll leave `config-toml-mode` and `dockerfile-mode` there for now, but I'm all good if we'd rather not add any dprint entries there for now.